pkg/chains/storage/gcs/gcs.go

```diff
@@ -181,18 +179,18 @@ func (b *Backend) retrieveObject(ctx context.Context, object string) (string, er
 	return string(payload), nil
 }
 
-func (b *Backend) sigName(opts config.StorageOpts) string {
-	return fmt.Sprintf(SignatureNameFormat, b.tr.Namespace, b.tr.Name, opts.Key)
+func (b *Backend) sigName(tr *v1beta1.TaskRun, opts config.StorageOpts) string {
+	return fmt.Sprintf(SignatureNameFormat, tr.Namespace, tr.Name, opts.Key)
```

Comment:
here and elsewhere: you no longer need `Backend` anymore to implement these.

```suggestion
func sigName(tr *v1beta1.TaskRun, opts config.StorageOpts) string {
	return fmt.Sprintf(SignatureNameFormat, tr.Namespace, tr.Name, opts.Key)
```

Reply:
Good point! 👍